### PR TITLE
✨ long running creation of studies

### PR DESCRIPTION
**Pull request by sanderegg**
## What do these changes do?

This PR leverages the new framework for long running REST calls when creating new study in oSparc.
Until now the creation of a study (and its copy with or without data) were done by the webclient calling ```POST /projects``` on the webserver. The latter would create a new study, maybe copy data and return a 201 code.

This is now changed to
- client calls ```POST /projects``` with same query parameters
- webserver answers very fast with a 202 and start creating the new study in the background.
- client polls ```GET /tasks/{task_id}``` to find out when the creation completed
- client gets the new project by calling ```GET /tasks/{task_id}/result```, server will return a 201 code.

needs #3260 

## Related issue/s

help with https://github.com/ITISFoundation/osparc-issues/issues/558

### CODE REVIEW

File: api/specs/webserver/openapi-tasks.yaml [hunk 1]
```diff
@@ -0,0 +1,70 @@
+paths:
+  /tasks:
+    get:
+      operationId: list_tasks
+      tags:
+        - tasks
+      responses:
+        "200":
+          description: Lists all the tasks currently available/running
```
Comment (GitHK) [inline api/specs/webserver/openapi-tasks.yaml:9]: I would have preferred if this was filtered by user_id.

Comment (GitHK) [reply api/specs/webserver/openapi-tasks.yaml:9]: Question: where was this implemented, I did not see it. Was it a previous pull request?

**Review by GitHK — COMMENTED** [review]
Looking nice, please find some questions below.

File: services/web/server/src/simcore_service_webserver/projects/projects_handlers_crud.py [hunk 2]
```diff
@@ -116,153 +122,214 @@ class Config:
...
+        forced_copy_project_id=None,
+        clean_output_data=(query_params.copy_data == False),
+    )
+    # remove template/study access rights
+    new_project["accessRights"] = {}
+    if not query_params.as_template:
+        new_project["name"] = default_copy_project_name(source_project["name"])
+    # the project is to be hidden until the data is copied
+    query_params.hidden = query_params.copy_data
+    # TODO: this should be made long running as well
```
[the remark this answers is not in the document]
Comment (sanderegg) [reply services/web/server/src/simcore_service_webserver/projects/projects_handlers_crud.py:181]: I do not understand that comment. are you talking about the TODO there? what would a context manager be useful for here? Anyway the server side must implement it first.

File: services/web/server/tests/unit/with_dbs/02/test_projects_cancellations.py [hunk 3]
```diff
@@ -49,6 +50,11 @@ def standard_user_role_response() -> tuple[
     )
 
 
+@pytest.mark.skip(
+    reason="since long running tasks are now used for copying, this is no longer an issue."
+    "If the timeout is really super short, then the background task should be properly cancelled."
+    "this should be tested now"
+)
```
[the remark this answers is not in the document]
Comment (sanderegg) [reply services/web/server/tests/unit/with_dbs/02/test_projects_cancellations.py:57]: yes, I had forgotten that one. I need to adapt the test.

File: services/web/server/src/simcore_service_webserver/projects/projects_handlers_crud.py (hunk 2 above)
[the remark this answers is not in the document]
Comment (GitHK) [reply services/web/server/src/simcore_service_webserver/projects/projects_handlers_crud.py:181]: Updated comment, OK I missed the fact that the server side was lacking

**Review by GitHK — APPROVED** [review]
👍 It's fine with me.

**Review by odeimaiz — APPROVED** [review]

File: services/api-server/src/simcore_service_api_server/modules/webserver.py [hunk 4]
```diff
@@ -114,8 +123,29 @@ async def create_project(self, project: NewProjectIn):
             ),  ## FIXME: REEAAAALY HACKY!
             cookies=self.session_cookies,
         )
-
         data: Optional[JSON] = self._process(resp)
+        assert data  # nosec
+        assert isinstance(data, dict)
```
Comment (pcrespov) [inline services/api-server/src/simcore_service_api_server/modules/webserver.py:128]: nosec?

File: services/api-server/src/simcore_service_api_server/modules/webserver.py [hunk 5]
```diff
@@ -114,8 +123,29 @@ async def create_project(self, project: NewProjectIn):
...
+        status_url = data["status_href"].lstrip(f"/{self.vtag}")
+        result_url = data["result_href"].lstrip(f"/{self.vtag}")
```
Comment (pcrespov) [inline services/api-server/src/simcore_service_api_server/modules/webserver.py:130]: why lstrip?

File: services/web/server/src/simcore_service_webserver/application.py [hunk 6]
```diff
@@ -62,6 +64,7 @@ def create_application() -> web.Application:
 
     # core modules
     setup_app_tracing(app)  # WARNING: must be UPPERMOST middleware
+    setup_long_running_tasks(app, router_prefix=f"/{API_VTAG}/tasks")
```
Comment (pcrespov) [inline services/web/server/src/simcore_service_webserver/application.py:67]: put is below rest

**Review by pcrespov — APPROVED** [review]

File: packages/service-library/src/servicelib/aiohttp/long_running_tasks/_routes.py [hunk 7]
```diff
@@ -31,12 +32,18 @@ async def get_task_result(request: web.Request) -> web.Response:
...
-    # NOTE: this will fail if the task failed for some reason....
-    await tasks_manager.remove_task(path_params.task_id, reraise_errors=False)
-    return task_result
+    # NOTE: this might raise an exception that will be catched by the _error_handlers
+    try:
+        task_result = tasks_manager.get_task_result(task_id=path_params.task_id)
+        # NOTE: this will fail if the task failed for some reason....
+        await tasks_manager.remove_task(path_params.task_id, reraise_errors=False)
+        return task_result
+    except (TaskNotFoundError, TaskNotCompletedError):
```
[the remark this answers is not in the document]
Comment (sanderegg) [reply packages/service-library/src/servicelib/aiohttp/long_running_tasks/_routes.py:41]: handled by _error_handlers.py used as a middleware

File: services/web/server/src/simcore_service_webserver/application.py (hunk 6 above)
Comment (sanderegg) [reply services/web/server/src/simcore_service_webserver/application.py:67]: we decided to leave it for now.

**Pull request merged**